nucliadb_node/src/writer/mod.rs

```diff
@@ -241,4 +247,54 @@ impl NodeWriterService {
             .collect();
         ShardIds { ids }
     }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn move_shard(
+        &self,
+        shard_id: &ShardId,
+        address: impl ToSocketAddrs + Clone,
+    ) -> ServiceResult<Option<()>> {
+        let Some(shard) = self.get_shard(shard_id) else {
+            return Ok(None);
+        };
+
+        Publisher::default()
+            .append(&shard.path)
+            // `unwrap` call is safe since the shard path already terminate by a valid file name.
+            .unwrap()
+            .retry_on_failure(RetryPolicy::MaxDuration(MAX_MOVE_SHARD_DURATION))
+            .send_to(address)
+            .await
+            .map(Some)
+            .map_err(|e| match e {
+                nucliadb_ftp::Error::IoError(e) => ServiceError::IOErr(e),
+                _ => ServiceError::GenericErr(Box::new(e)),
+            })
+    }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn accept_shard(
+        &mut self,
```

Comment:
This worries me, we are blocking the reader to accept the shard. I guess because of the `load_shard` call. That is why I think the logic of `accepting` a shard should be at on the grpc driver, before taking the service lock.